AddonExample/AddonExampleEventSource.cs

```diff
@@ -0,0 +1,177 @@
+﻿using Newtonsoft.Json.Linq;
+using RainbowMage.OverlayPlugin;
+using System;
+using System.Collections.Generic;
+using System.Windows.Forms;
+
+namespace AddonExample
+{
+
+    public class AddonExampleEventSource : EventSourceBase, ILogger
+    {
+        public AddonExampleEventSourceConfig Config { get; private set; }
+
+        // Original Timer
+        System.Timers.Timer originalTimer;
+
+        // Events
+        public delegate void AddonExampleOriginalTimerFiredHandler(JSEvents.OriginalTimerFiredEvent e);
+        public event AddonExampleOriginalTimerFiredHandler OnAddonExampleOriginalTimerFired;
+
+        public delegate void AddonExampleEmbeddedTimerFiredHandler(JSEvents.EmbeddedTimerFiredEvent e);
+        public event AddonExampleEmbeddedTimerFiredHandler OnAddonExampleEmbeddedTimerFired;
+
+
+        public AddonExampleEventSource(RainbowMage.OverlayPlugin.ILogger logger) : base(logger)
+        {
+            Name = "AddonExampleES";
+
+            // Register Events subscribe to other EventSources/Overlays
+            RegisterEventTypes(new List<string>()
+            {
+                "onAddonExampleOriginalTimerFiredEvent", "onAddonExampleEmbeddedTimerFiredEvent",
+            });
+
+
+            // Register EventHandler
+            // This EventHandler is called from other EventSources/Overlays
+            // You can execute some process or response data.
+            
+            RegisterEventHandler("addonExampleSay", (msg) => {
+                Advanced_Combat_Tracker.ActGlobals.oFormActMain.TTS(msg["text"].ToString());
+                return null;
+            });
+            RegisterEventHandler("addonExampleCurrentTime", (msg) => {
+                var ret = new JObject();
+                ret["time"] = DateTimeOffset.UtcNow.ToString();
+                return ret;
+            });
+        }
+
+        public override Control CreateConfigControl()
+        {
+            return new AddonExampleEventSourceConfigPanel(this);
+        }
+
+        public override void LoadConfig(IPluginConfig config)
+        {
+            Config = AddonExampleEventSourceConfig.LoadConfig(config);
+        }
+
+        public override void SaveConfig(IPluginConfig config)
+        {
+            Config.SaveConfig(config);
+        }
+
+        public override void Start()
+        {
+            // Reset Event Dispatcher
+            OnAddonExampleOriginalTimerFired -= (e) => DispatchToJS(e);
+            OnAddonExampleOriginalTimerFired += (e) => DispatchToJS(e);
+            OnAddonExampleEmbeddedTimerFired -= (e) => DispatchToJS(e);
+            OnAddonExampleEmbeddedTimerFired += (e) => DispatchToJS(e);
```

Comment:
Why do you remove and add the event handlers here? Does that change anything?

Reply:
I noticed that there was no need to use Event/Delegate.
Deleted them by 2c281aa